scripts/sbatch.py

```diff
@@ -0,0 +1,288 @@
+        if line.startswith("PartitionName={}".format(partition_name)):
+            # PartitionName=prepost    Nodes=wbl[001-005,008-011] TRESBillingWeights="CPU=1.0,Mem=0.1429G"  QOS=p_prepost2 MaxTime=03:00:00 PriorityTier=1
+            try:
+                return get_billing_weights(line)
+            except (IndexError, ValueError):
+                # TRESBillingWeights not found, use default cpu_weights and mem_weights
+                for l in lines[::-1]:
+                    if l.startswith("PartitionName=DEFAULT"):
+                        return get_billing_weights(l)
+
+
+def calculate_requested_chours(
+    cpu_billing_weights,
+    mem_billing_weights,
+    mem_per_cpu,
+    ntasks=DEFAULT_NTASKS,
+    cpus_per_task=DEFAULT_CPUS_PER_TASK,
+):
+    """Calculate requested core hours for a job on mahuika"""
+    # https://slurm.schedmd.com/tres.html
+    # see above link for calculating formula
+    total_cpus = ntasks * cpus_per_task
+    total_mem = total_cpus * mem_per_cpu
+
+    requested_hours = (total_cpus * cpu_billing_weights) + (
+        total_mem * mem_billing_weights
+    )
+
+    return requested_hours
+
+
+def get_available_chours(json_file, account, username):
+    """
+       Read the core hours json file that's populated from the dashboard.db and
+       return the available core hours for a specified user in specified account
+    """
+    with open(json_file, "r") as f:
+        json_array = json.load(f)
+    for d in json_array:
+        if d.get(account) is not None:
+            for user_dict in d[account]:
+                if user_dict.get(username) is not None:
+                    return (
+                        user_dict[username]["allocation"] - user_dict[username]["used"]
+                    )
+                else:
+                    sys.exit(
+                        "No core hours usage info for {} {} from json file {}".format(
+                            account, username, json_file
+                        )
+                    )
+        else:
+            sys.exit(
+                "No core hours usage info for {} from json file {}".format(
+                    account, json_file
+                )
+            )
+
+
+def compare_hours_and_sbatch(requested_hours, available_hours, hpc, sl_to_sbatch):
+    """
+       Compare requested hour with available hours
+       and either submit the job or reject
+    """
+    if requested_hours <= available_hours:
+        cmd = "{} -M {} {}".format(SBATCH_BIN, hpc, sl_to_sbatch)
+        subprocess.call(cmd, shell=True)
+    else:
+        sys.exit("Not enought core hours left, please contact Jonney")
+
+
+def process_slurm_header(sl_file):
+    """Read a slurm file and returns the wanted header values as a dictionary"""
+    with open(sl_file) as f:
+        lines = f.readlines()
+    # SBATCH --account=nesi00213
+    header_dict = {h.value: None for h in SlurmHeader}
+    for header in header_dict.keys():
+        for line in lines:
```

Comment:
Iterates over the entire slurm file {header} times. Better off doing the iteration the other way around?

Reply:
Another thing to note, is that once we added the ability to parse additional arguments for slurm itself ( as stated in another comment), we will need to check if these parameters are supplied in the args, instead of a header in *.sl